pkg/common-controller/snapshot_controller_base.go

```diff
@@ -243,7 +246,7 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 			return false
 		}
 		newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
-		if err == nil {
+		if err == nil || errors.IsNotFound(err) {
```

Comment:
Can you add a note here to say that we still need to delete snapshot if snapshot class is not found?

Reply:
I've included a note here. Should there be a log message printed as well? Debug level?

Reply:
Yes

Reply:
I've broken this conditional into two separate statements and provided different messages for each.